File: examples/python/documentation/main-remote-backend.py
from cdktf import CloudBackend, NamedCloudWorkspace


# DOCS_BLOCK_START:remote-backend-define
from constructs import Construct
from cdktf import App, CloudBackend, NamedCloudWorkspace,TerraformStack, TerraformOutput

class MyStack(TerraformStack):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        CloudBackend(self,
            hostname = "app.terraform.io",
            organization = "company",
            workspaces = NamedCloudWorkspace("my-app-prod")
        )

        TerraformOutput(self, "dns-server",
            value = "hello-world"
        )

app = App()
MyStack(app, "hello-terraform")
app.synth
# DOCS_BLOCK_END:remote-backend-define

# DOCS_BLOCK_START:remote-backend-migrate
stack = MyStack(app, "hi-terraform")
CloudBackend(stack,
    hostname = "app.terraform.io",
    organization = "company",
    workspaces = NamedCloudWorkspace("my-app")
)
# DOCS_BLOCK_END:remote-backend-migrate

# DOCS_BLOCK_START:remote-backend-escape-hatches
stack.add_override("terraform.backend",{
    "atlas": {
        "name": "example_corp/networking-prod",
        "address": "https://app.terraform.io"
    }
})
# DOCS_BLOCK_END:remote-backend-escape-hatches

File: examples/java/documentation/src/main/java/com/mycompany/app/remoteBackends/MainRemoteBackend.java
package com.mycompany.app.remoteBackends;

import com.hashicorp.cdktf.*;
import software.constructs.Construct;

import java.util.HashMap;

// DOCS_BLOCK_START:remote-backend-migrate
public class MainRemoteBackend extends TerraformStack {

    public MainRemoteBackend(Construct scope, String id) {
        super(scope, id);

        new TerraformOutput(this, "dns-server", TerraformOutputConfig.builder()
                .value("local")
                .build()
        );
    }

    public static void main(String[] args) {
        final App app = new App();
        MainRemoteBackend stack = new MainRemoteBackend(app, "local-to-cloud-backend");
        new CloudBackend(stack, CloudBackendConfig.builder()
                .hostname("app.terraform.io")
                .organization("company")
                .workspaces(new NamedCloudWorkspace("my-app-prod"))
                .build()
        );
        app.synth();
        // DOCS_BLOCK_END:remote-backend-migrate

        // DOCS_BLOCK_START:remote-backend-escape-hatches
        stack.addOverride("terraform.backend", new HashMap<String, HashMap>() {
            {
                put("atlas", new HashMap<String, String>() {
                    {
                        put("name", "example_corp/networking-prod");
                        put("address", "https://app.terraform.io");
                    }
                });
            }
        });
        // DOCS_BLOCK_END:remote-backend-escape-hatches
        // DOCS_BLOCK_START:remote-backend-migrate
    }
}
// DOCS_BLOCK_END:remote-backend-migrate
